File: C++/3list/Pentagon.cpp
/*
@author: Stanisław Woźniak
@title: pentagon.cpp
*/

#include "Pentagon.hpp"
#include <cmath>

Pentagon::Pentagon(double side){
  this -> side = side;
}

double Pentagon::field(){
  return side*side*1.25*tan((54*M_PI)/180);
}

double Pentagon::perimeter(){
  return 5*side;
}

File: java/3list/src/com/stanislaw/moreFigures/Pentagon.java
package com.stanislaw.moreFigures;
/*
@author: Stanisław Woźniak
@version: 1.0
*/
import com.stanislaw.core.FigureImpl;

public class Pentagon extends FigureImpl {

    private double side;

    public Pentagon(double side) {
        this.side = side;
    }

    @Override
    public double field() {
        return this.side*this.side*1.25*Math.tan(Math.toRadians(54));
    }

    @Override
    public double perimeter() {
        return 5*this.side;
    }
}
